source/common/router/config_impl.cc

```diff
@@ -136,6 +137,37 @@ Optional<uint64_t> HashPolicyImpl::generateHash(const std::string& downstream_ad
   return hash;
 }
 
+std::vector<MetadataMatchCriterionConstSharedPtr>
+MetadataMatchCriteriaImpl::extractMetadataMatchCriteria(const MetadataMatchCriteriaImpl* parent,
+                                                        const ProtobufWkt::Struct& matches) {
+  std::vector<MetadataMatchCriterionConstSharedPtr> v;
+  std::unordered_map<std::string, std::size_t> existing;
```

Comment:
Size seems to be providing some notion of initial ordering in the loop that populates it below, and is then used to find the place in `v` that a parent entry would be, but a comment to explain would help.